### DISCUSSION TITLE
odoo-bin scaffolding with doodba development container

### DISCUSSION THREAD

**Discussion by nimarosa** [Tecnativa/doodba#423] [Q&A]
Okay so i'm trying to use odoo-bin to scaffold addons, but because the file really is in a container, the script fails. Do you know how i can achieve this? Maybe some docker-compose command to run the script? Because i'm familiar to integrating odoo scaffolding with VSCode, so it's a function that i really hope i can have without having to have a local install of odoo. 

Thank you for the great image, it's the best.

**Comment by joao-p-marques** [comment]
Hi :wave: 

Well, this is a bit tricky because of the way the code is mounted and handled in the container in the devel environment.
With https://github.com/Tecnativa/doodba-copier-template we mount the code as read only in the container in `/opt/odoo/custom/src` and then symlink all the addons to `opt/odoo/auto/addons`, where the container already has write permissions, so you could do:
1. Change the mount of `/opt/odoo/custom/src` to `rw` and generate the addon there
2. Generate the addon in `/opt/odoo/auto` and then copy it from outside the container to wherever you want
3. Generate the addon anywhere in the container and copy it outside with a `docker cp`

It would be nice to have a script to do this quickly. WDYT @Yajo? Maybe an invoke task?

**Comment by yajo** [comment]
I just copy-paste https://github.com/OCA/maintainer-tools/tree/master/template/module and remove/rename what I don't need, and it's more than enough for most cases.

If you really want something fancy, you can do an odoo module copier template.

**Reply by ndemarco** [reply]
Hello all, Confirming this still Doodba's suggested method to begin a new module. There's nothing wrong with it. Just seeing if there's an undocumented newer practice.

**Reply by yajo** [reply]
We use https://gitlab.com/moduon/odoo-module-template
